gsl: update to 2.7.1, adopt.

tornaria opened the pull request:
The main change is a soname bump that apparently should have been done on 2.7.

This requires several revbumps. I took the chance to:
 - giac: update to 1.7.0.43 (minor update, no surprises)
 - dieharder: fix URL, since it was failing with a certificate error
 - gnuradio: add missing checkdepends and patch from upstream so `do_check()` succeeds
 - guvcview and siril: trivial fixes so `do_check()` succeeds
 - krita: disable tests since they are broken (70 failures)

I tested pkg -Q on x86_64 and all passes with these fixes.

ericonr commented:
```
aarch64-linux-gnu-g++: fatal error: cannot read spec file 'libgomp.spec': No such file or directory
compilation terminated.
```

Your cross builds might be hit by #31055

tornaria commented:
> ```
> aarch64-linux-gnu-g++: fatal error: cannot read spec file 'libgomp.spec': No such file or directory
> compilation terminated.
> ```
> 
> Your cross builds might be hit by #31055

Hum... what shall I do?

The results of the CI are kind of weird; the one for x86_64 finished ok the first time and now it failed. Others failed because a download error (?). I worked hard to fix `do_check()` in four packages so this PR would work :thinking: 

We discussed with @leahneukirchen on IRC how to setup CI so that it is possible to run xbps-src pkg without -Q on cases like this one:

```
10:31 < leah2> gt7: unfortunately not. i always wanted two actions, first build then 
               check, but it seems not possible to implement efficiently?
10:43 < gt7> leah2: maybe an option like [ci-skip] but [check-skip] which will still 
             run ./xbps-src pkg but without -Q ?
10:43 < leah2> yeah, could work
10:45 < gt7> I don't know anything about how github actions work, but the line
10:45 < gt7> TEST: '${{ matrix.config.test }}'
10:46 < gt7> if that can be made conditional to finding [check skip] on the PR title 
             or body, that may work
10:47 < gt7> it should be enough to have an expression that normally evaluates to 
             matrix.config.test, but yields 0 if [check skip] is present in the PR
10:48 -void-robot:#xbps- leahneukirchen pushed to void-packages (stress-ng: update to 
          0.13.08.)
10:49 < gt7> in fact looking at common/travis/build.sh, anything different than 1 
             works so maybe "[skip]" is better so that one can see at the matrix that 
             check was skipped by request
10:50 < gt7> I mean: set TEST to "[skip]" instead of 0 when "[check skip]" is 
             pressent in the PR
```

at least it would save time.

leahneukirchen commented:
The gomp issue should not affect building as it will happen in a clean masterdir.

tornaria commented:
I rebased to current master (manual intervention needed due to inkscape update).

This is my first update with so many dependencies. What's the way to move forward from here?

ericonr commented:
I restarted the build to see what happened with gnuradio, since it failed a check on glibc.

I can test some of these and cross builds locally so we can merge this regardless of CI.

tornaria commented:
> I can test some of these and cross builds locally so we can merge this regardless of CI.
I'm now running `xbulk -Q $(xrevshlibs gsl)` on a x86_64 masterdir. I'll also run on x86_64-musl, on i686, and on cross aarch64.

I'll report back when it's done.

tornaria commented:
Result of `./xbps-src pkg -Q` so far:

On x86_64:
- [x] gsl
- [x] Clight
- [x] bogofilter
- [x] krita
- [x] calligra
- [x] dieharder
- [x] enblend-enfuse
- [x] giac
- [ ] gnuradio: **check fails on avx512, patch available upstream, will retest**
- [x] guvcview
- [x] igt-gpu-tools
- [x] inkscape
- [x] ipe
- [x] siril

On x86_64-musl:
- [x] gsl
- [x] Clight
- [ ] bogofilter: **3 tests fail**
- [x] krita
- [ ] calligra
- [ ] dieharder
- [ ] enblend-enfuse
- [ ] giac
- [ ] gnuradio
- [ ] guvcview
- [ ] igt-gpu-tools
- [ ] inkscape
- [ ] ipe
- [ ] siril

On i686:
- [x] gsl
- [x] Clight
- [x] bogofilter
- [x] krita
- [x] calligra
- [x] dieharder
- [x] enblend-enfuse
- [x] giac
- [ ] gnuradio: **5 tests fail**
- [ ] guvcview
- [ ] igt-gpu-tools
- [ ] inkscape
- [ ] ipe
- [ ] siril

On cross aarch64:
- [x] gsl
- [x] Clight
- [x] bogofilter
- [x] krita
- [x] calligra
- [x] dieharder
- [x] enblend-enfuse
- [x] giac
- [ ] gnuradio
- [ ] guvcview
- [ ] igt-gpu-tools
- [ ] inkscape
- [ ] ipe
- [ ] siril

ericonr commented:
inkscape failures on musl are expected, known, and reported upstream. I'm just waiting for them to be fixed because idk my way around the codebase :shrug: 

Thanks for getting started on the builds, I can't do a whole lot of builds until the 30th because I'm on metered internet :P

tornaria commented:
To summarize, everything ok with pkg -Q except:
 - `bogofilter` on musl: 3 tests fail (t.dump.load, t.nonascii.replace, t.bulkmode)
 - `inkscape` on musl: 13 tests fail (all with SEGFAULT)
 - `gnuradio` on i686: 5 tests fail (qa_pmt, qa_fecapi_ldpc, qa_agc, qa_ctcss_squelch, qa_constellation)

NOTE: for gnuradio on i686 the test `qa_fecapi_ldpc` seems to hang forever (I killed after half an hour and I think that's what timeouts CI after 6 hours).

In my rebase above I included a commit that is named "DO NOT MERGE" which is an attempt of a workaround to #31055 to see if we can have CI pass on cross... It's just adding `libgomp-devel` to `makedepends`, I wonder if merging it would cause any harm.

tornaria commented:
The last CI:
 - x86_64 and aarch64 pass all
 - i686 hangs on test `qa_fecapi_ldpc` as expected (3h30 stuck at that until github kills it)
 - x86_64-musl fails on check inkscape as expected (it was either that one or bogofilter that would fail)
 - the other cross builds don't fail, they just run out of time (note aarch64 just barely made it in 344m while the limit is 360m).

A fix for test qa_fecapi_ldpc on gnuradio so it doesn't hang would be very desirable, but the others... I'm tired of this (I only intended to update gsl and I already fixed several issues on other pkgs :thinking:).

----

Random ideas:
 - maybe we should fill issues for failing tests, so it's documented (e.g. when revbump, to know it's not an issue introduced by the lib update)
 - maybe have a way to indicate that some checks are expected to fail for certain archs. For instance, I'd mark bogofilter and inkscape as xfail for x86_64-musl. The check step would still run, but it wouldn't stop with an error on failure so the CI can keep going. E.g. `check_xfail="x86_64-musl"`.
 - maybe have a way to indicate that some checks should be skipped for certain archs. For instance, I'd mark gnuradio to skip checks for i686 so the CI wouldn't get stuck on a loose test. E.g. `check_skip="i686".

Policy could be to comment any `check_xfail` or `check_skip` with an issue number where it is well documented.

More random thoughts:
 - maybe useful to have an option for CI such that a failure on check doesn't stop; the pkg step would be run anyway. The whole thing would still show a failure at the end but at least one would know the packaging step runs ok.
 - similarly, if one package fails, keep building the rest (as long as its dependencies didn't fail on build or pkg). At the end print a complete and concise report at the end about which steps (build/check/pkg) failed for which pkgs, and still give error if anything failed.
 - I wonder if it would be possible to have "dynamic" steps in a CI job, I mean: instead of having one step "Build and check packages" have one separate step for each pkg that is built so it's easy to see logs for each pkg individually
 - Another question: would -t work in the CI? Would it make sense to run using it so building of one pkg doesn't affect building of others (avoid the libgomp issue). If -t doesn't work, could zap the masterdir between pkgs.

tornaria commented:
About the hang on gnuradio, it seems to be known and still open (https://github.com/gnuradio/gnuradio/issues/989)

An alternative would be to disable the particular tests (mark them as skip or xfail) for bogofilter, inkscape and gnuradio conditional to architecture. That depends on the particular build system and my suggestion above is to avoid wrestling with them. If anybody knows an easy way to mark tests as skip or xfail on those three packages, we could do that.

tornaria commented:
Failures for gnuradio on i686:
```
98% tests passed, 5 tests failed out of 228

Total Test time (real) = 1860.47 sec

The following tests FAILED:
         20 - qa_pmt (Failed)
        111 - qa_fecapi_ldpc (Failed)
        149 - qa_agc (Failed)
        151 - qa_ctcss_squelch (Failed)
        176 - qa_constellation (Failed)
```
NOTES:
 - I killed qa_fecapi_ldpc as it was looping forever (see https://github.com/gnuradio/gnuradio/issues/989)
 - qa_agc may be fixed by a patch I added later that fixed that test for x86_64

DETAILS:
```
 20/228 Test  #20: qa_pmt .......................................***Failed    0.20 sec
..............E..E.
======================================================================
ERROR: test15 (__main__.test_pmt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/pmt/qa_pmt.py", line 128, in test15
    x_pmt = pmt.from_long(const)
  File "/builddir/gnuradio-3.8.0.0/build/gnuradio-runtime/python/pmt/../../swig/pmt_swig.py", line 1942, in from_long
    return _pmt_swig.from_long(x)
OverflowError: in method 'from_long', argument 1 of type 'long'

======================================================================
ERROR: test18 (__main__.test_pmt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/pmt/qa_pmt.py", line 153, in test18
    x_pmt = pmt.from_long(const)
  File "/builddir/gnuradio-3.8.0.0/build/gnuradio-runtime/python/pmt/../../swig/pmt_swig.py", line 1942, in from_long
    return _pmt_swig.from_long(x)
OverflowError: in method 'from_long', argument 1 of type 'long'

----------------------------------------------------------------------
Ran 19 tests in 0.002s
       
FAILED (errors=2)
```

```
111/228 Test #111: qa_fecapi_ldpc ...............................***Failed  1573.24 sec
/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr/top_block.py:59: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  self.setDaemon(1)
...Terminated
```

```
149/228 Test #149: qa_agc .......................................***Failed    1.25 sec
/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr/top_block.py:59: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  self.setDaemon(1)
.........F..
======================================================================
FAIL: test_006 (__main__.test_agc)
Test the complex AGC loop (attack and decay rate inputs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gr-analog/python/analog/qa_agc.py", line 486, in test_006
    self.assertFloatTuplesAlmostEqual(result, M*[ref,], 4)
  File "/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr_unittest.py", line 110, in assertFloatTuplesAlmostEqual
    self.assertEqual(len(a), len(b))
AssertionError: 96 != 100

----------------------------------------------------------------------
Ran 12 tests in 0.007s

FAILED (failures=1)
```

```
151/228 Test #151: qa_ctcss_squelch .............................***Failed    1.25 sec
./builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr/top_block.py:59: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  self.setDaemon(1)
F.
======================================================================
FAIL: test_ctcss_squelch_002 (__main__.test_ctcss_squelch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gr-analog/python/analog/qa_ctcss_squelch.py", line 82, in test_ctcss_squelch_002
    self.assertFloatTuplesAlmostEqual(expected_result, result_data, 4)
  File "/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr_unittest.py", line 110, in assertFloatTuplesAlmostEqual
    self.assertEqual(len(a), len(b))
AssertionError: 39 != 29

----------------------------------------------------------------------
Ran 3 tests in 0.002s

FAILED (failures=1)
```

```
176/228 Test #176: qa_constellation .............................***Failed   10.72 sec
/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr/top_block.py:59: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  self.setDaemon(1)
.FF.
======================================================================
FAIL: test_soft_qam16_calc (__main__.test_constellation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gr-digital/python/digital/qa_constellation.py", line 319, in test_soft_qam16_calc
    self.assertFloatTuplesAlmostEqual(y_cpp_raw_calc, y_cpp_table, 0)
  File "/builddir/gnuradio-3.8.0.0/gnuradio-runtime/python/gnuradio/gr_unittest.py", line 110, in assertFloatTuplesAlmostEqual
    self.assertEqual(len(a), len(b))
AssertionError: 17 != 25

======================================================================
FAIL: test_soft_qpsk_calc (__main__.test_constellation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/gnuradio-3.8.0.0/gr-digital/python/digital/qa_constellation.py", line 279, in test_soft_qpsk_calc
    self.assertEqual(y_cpp_raw_calc, y_cpp_table)
AssertionError: Lists differ: [0.0,[16 chars]1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1[13 chars] 1.0] != [0.0,[16 chars]1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1[28 chars] 1.0]

First differing element 5:
1.0
0.0

Second list contains 3 additional elements.
First extra element 17:
0.0

  [0.0,
   0.0,
   1.0,
   0.0,
   1.0,
+  0.0,
   1.0,
+  0.0,
+  1.0,
+  0.0,
+  1.0,
+  0.0,
   0.0,
   1.0,
   1.0,
   0.0,
-  0.0,
   1.0,
   0.0,
-  1.0,
-  1.0,
   0.0,
   1.0]

----------------------------------------------------------------------
Ran 4 tests in 9.216s

FAILED (failures=2)
```

tornaria commented:
Failures for bogofilter on x86_64-musl:
```
...
FAIL: t.dump.load
FAIL: t.nonascii.replace
...
FAIL: t.bulkmode
...
============================================================================
Testsuite summary for bogofilter 1.2.5
============================================================================
# TOTAL: 63
# PASS:  57
# SKIP:  3
# XFAIL: 0
# FAIL:  3
# XPASS: 0
# ERROR: 0
```

tornaria commented:
Failures for inkscape on x86_64-musl:
```
96% tests passed, 13 tests failed out of 299

Total Test time (real) = 179.25 sec
...
The following tests FAILED:
          3 - test_drag-and-drop-svgz (SEGFAULT)
          6 - test_color-profile (SEGFAULT)
          8 - test_sp-object (SEGFAULT)
          9 - test_object-set (SEGFAULT)
         10 - test_object-style (SEGFAULT)
         12 - test_rebase-hrefs (SEGFAULT)
         13 - test_style-elem (SEGFAULT)
         20 - test_sp-gradient (SEGFAULT)
         22 - test_object (SEGFAULT)
         24 - test_cairo-utils (SEGFAULT)
         25 - test_svg-extension (SEGFAULT)
         28 - test_lpe-bool (SEGFAULT)
         30 - test_sp-item-group (SEGFAULT)
```

tornaria commented:
More patches for gnuradio. Now there's no hang on i686, but two tests still fail.

I'd merge as is (with or without the last commit; if merging the last commit should be squashed to the second-to-last one).

tornaria commented:
Updated summary: everything ok with pkg -Q except:
 - `bogofilter` on musl: 3 tests fail (t.dump.load, t.nonascii.replace, t.bulkmode)
 - `inkscape` on musl: 13 tests fail (all with SEGFAULT)
 - `gnuradio` on i686: 2 tests fail (qa_ctcss_squelch, qa_constellation)

CI is as good as it will get. Should it be merged @ericonr ?

tornaria commented:
Rebased to master (taking care to revbump inkscape and krita again) and updated giac to 1.7.0.45.

@ericonr: do you think anything else is needed here?

tornaria commented:
After looking at siril source code, and https://github.com/void-linux/void-packages/issues/31055#issuecomment-1008935998, I think adding `libgomp-devel` unconditionally to siril depends may be a good thing to do.
